### PR TITLE
Add a pre-commit setup to the repository. 

### PR DESCRIPTION
**Pull request by vdemeester**
# Changes

This will allow us to run some checks before committing. If `direnv` is enabled in the developer environment *and* `pre-commit` available, it will automatically install the hooks for the repository.

~~On top of #2175.~~

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Run the code checkers with `make check`
- [ ] Regenerate the manpages, docs and go formatting with `make generated`
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/cli/blob/master/CONTRIBUTING.md)
for more details._

# Release Notes

```release-note
NONE
```

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
+default_install_hook_types: ["pre-push"]
+exclude: '(^docs/themes/hugo-book|^vendor|.*golden$|^\.vale)'
```
[the remark this answers is not in the document]
Comment (vdemeester) [reply .pre-commit-config.yaml]: ah crap, I forgot to remove that part 😹

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,38 @@
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
+default_install_hook_types: ["pre-push"]
+exclude: '(|^vendor|.*golden$)'
```
Comment (chmouel) [inline .pre-commit-config.yaml:4]:
```suggestion
exclude: '(^vendor|.*golden$)'
```

**Comment by chmouel** [comment]
I am good with the pre-commit part but still have concerns about .env file

**Comment by chmouel** [comment]
Thanks for the update without dotenv,

/lgtm

**Comment by roman-kiselenko** [comment]
I guess the note about [pre-commit](https://pre-commit.com/) installation should be in the https://github.com/tektoncd/cli/blob/main/DEVELOPMENT.md#requirements 🤔

**Comment by chmouel** [comment]
@roman-kiselenko that would be a great in this document have a brief explanation about what pre-commit does and how to install the hooks (basically run pre-commit binary at the root directory)

maybe mentions this as well in the [pull request template](https://github.com/tektoncd/cli/blob/ac5df8b1596efb18e750b9fe02191e1ab0c744f3/.github/pull_request_template.md?plain=1#L15) with a link to that doc

**Pull request merged**